packages/main/src/plugin/locked-configuration.ts

```diff
@@ -0,0 +1,57 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+
+import { inject, injectable } from 'inversify';
+
+import { Directories } from './directories.js';
+import { SYSTEM_LOCKED_FILENAME } from './managed-by-constants.js';
+
+@injectable()
+export class LockedConfiguration {
```

Comment:
IHMO it's confusing that we have a LockedConfiguration that read the locked file and all documentation is in configuration-impl while it applied here

and why getLockedConfigValue do not belong to this object ?